_episodes_rmd/05-r-markdown.Rmd

```diff
@@ -177,169 +241,216 @@ cat('```{r packages}\nlibrary(tidyverse)\n```\n')
 
-maxGDP <- gapminder_1997 %>%
-  summarise(round(max(gdpPercap))) %>%
-  pull()
-```
+> ## Code chunk options
+> Which of the following would lead to a report with no code chunks? For the ones that would not work, what would happen instead?
+> 1. Add `echo = FALSE` inside the curly brackets of the plotting code chunk.
+> 1. Add `include = FALSE` inside the curly brackets of the plotting code chunk.
+> 1. Add `echo = FALSE` inside the curly brackets of the setup code chunk.
+> 1. Change `echo = TRUE` to `echo = FALSE` in the `knitr::opts_chunk$set()` function in the first code chunk. 
+>
+> > ## Solution
+> > 1. This would only remove the code for the plotting code chunk, but not the packages code chunk. 
+> > 1. This would remove the code but also the plot from the report. 
+> > 1. This would not change anything because that code chunk is already excluded because `include = FALSE`.
+> > 1. This would remove all code from the output file (what we want). 
+> {: .solution}
+{: .challenge}
 
-Now, all we need to do is reference the values we just computed to describe our
+### Bonus 
+
+> ## Dynamically updating text using mini R chunks
+> Sometimes, you want to describe your data or results (like our plot) to the audience in text but the data and results may still change as you work things out. R Markdown offers an easy way to do this dynamically, so that the text updates as your data or results change. Here is how to do this.
+> 
+> Let's create a code chunk that summarizes features of our data that we can use to describe our plot to  our audience. Note that we set `include=FALSE` because we only want this step to happen in the background. For our purposes, we will calculate how many countries were included in the analysis, as well as the minimum and maximum percent of smokers.
+> > ## Calculating summary statistics
+> > Perform the following steps to calculate summary statistics:
+> > 1. Create a subsetted `smoking` datset including only data from 2019 an save it to the variable `smoking_2019`. 
+> > 1. Look up how to use the function `n_distinct()` and use it to find the number of distinct countries in `smoking_2019`. HINT: You can pick the column you're interested in using `select()` and then pipe that to `n_distinct()`. 
+> > 1. Find the minimum value of `smoke_pct` and round it to 1 decimal place.
+> > 1. Find the maximum value of `smoke_pct` and round it to 1 decimal place.
+> >
+> > > ## Solution
+> > > ```{r}
+> > > smoking_2019 <- smoking %>% 
+> > >   filter(year == 2019)
+> > > smoking_2019 %>%
+> > >   select(country) %>%
+> > >   n_distinct()
+> > > smoking_2019 %>%
+> > >   summarise(round(min(smoke_pct), 1))
+> > > smoking_2019 %>%
+> > >   summarise(round(max(smoke_pct), 1)) 
+> > > ```
+> > {: .solution}
+> {: .challenge}
+> Let's fix up the code you wrote in the previous section a bit to make it easier to use for our purposes. 
```

Comment:
To shorten, I think you can get rid of the section above, and provide a little more of an intro for this section.